types/stripe-v3/index.d.ts

```diff
@@ -165,6 +197,94 @@ declare namespace stripe {
         metadata: any;
         name?: string;
         tokenization_method?: tokenizationType;
+        three_d_secure?: 'required' | 'recommended' | 'optional' | 'not_supported';
+    }
+
+    interface RetrieveSourceOptions {
+        id: string;
+        client_secret: string;
+    }
+
+    // Container for all payment request related types
+    namespace paymentRequest {
+        interface StripePaymentRequestUpdateOptions {
+            currency: string;
+            total: {
+                amount: number;
+                label: string;
+                pending?: boolean;
+            };
+            displayItems?: Array<{
+                amount: number;
+                label: string;
+                pending?: boolean;
+            }>;
+            shippingOptions?: ShippingOption[];
+        }
+
+        interface StripePaymentRequestOptions extends StripePaymentRequestUpdateOptions {
+            country: string;
+            requestPayerName?: boolean;
+            requestPayerEmail?: boolean;
+            requestPayerPhone?: boolean;
+            requestShipping?: boolean;
+        }
+
+        interface UpdateDetails {
+            status: 'success' | 'fail' | 'invalid_shipping_address';
```

Comment:
It could be useful to have an enum type

Reply:
When you say an enum type do you mean a proper enum type with string literals or do you mean one like the ones written in the rest of this definition files (type aliases)? I didn't want to overload it with too many types, but I don't mind either way.

Reply:
Yep, I'd prefer an enum, not those type aliases